src/NLog/Targets/Wrappers/ConcurrentRequestQueue.cs

```diff
@@ -223,11 +224,10 @@ public override void DequeueBatch(int count, IList<AsyncLogEventInfo> result)
 
             if (dequeueBatch)
             {
-                bool lockTaken = Monitor.TryEnter(_logEventInfoQueue);    // Try to throttle
+                bool lockTaken = result.Count == count && Monitor.TryEnter(_logEventInfoQueue);    // Only try throttle when falling behind
```

Comment:
I think this is OK, but maybe rename lockTaken? 

proposal: needsPulse

Reply:
No it is `lockTaken`, because it needs to release it afterwards. Pulse is just a bonus-operation to be nice.

Reply:
ow yes you're right!